packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -37,58 +44,349 @@ import {Heading, Text} from '@react-spectrum/typography';
+      </Content>
+      <ButtonGroup>
+        <Button autoFocus variant="cta" onPress={close}>Acknowledge</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
 
 ## Events
 
-*If the component supports event props, include this section. Only cover the events that are important to the main functionality of the component.*
+For Dialogs, user defined callbacks should be chained with the DialogTrigger's `close` function in the press/click handler
+of the Dialog's action buttons. The following example alerts if the Dialog's save or cancel button is clicked.
+
+```tsx example
+function Example() {
+  let alertSave = (cb) => {
+    cb();
+    alert('Profile saved!');
+  }
+
+  let alertCancel = (cb) => {
+    cb();
+    alert('Profile not saved!');
+  }
+
+  return (
+    <DialogTrigger>
+      <ActionButton>Set Profile</ActionButton>
+      {(close) => (
+        <Dialog>
+          <Heading>Profile</Heading>
+          <Divider />
+          <ButtonGroup>
+            <Button autoFocus variant="cta" onPress={() => alertSave(close)}>Save</Button>
+            <Button variant="secondary" onPress={() => alertCancel(close)}>Cancel</Button>
+          </ButtonGroup>
+          <Content>
+            <Form>
+              <TextField label="Name" />
+              <Checkbox>Make private</Checkbox>
+            </Form>
+          </Content>    
+        </Dialog>
+      )}
+    </DialogTrigger>
+  );
+}
+```
+
+Additionally, DialogTrigger accepts an `onOpenChange` prop which is triggered whenever the Dialog is opened or closed. 
+For more information on the `onOpenChange` prop see [DialogTrigger page](TODO LINK HERE).
 
-## Validation
+### Dismissable Dialogs
 
-*If the component supports validation props, include this section.*
+Dismissable Dialogs support a `onDismiss` prop which is triggered whenever the Dialog's close button is clicked.
+Similar to non-dismissable dialogs, you must chain the DialogTrigger's `close` function with whatever callback you provide as
+`onDismiss`.
+
+```tsx example
+function Example() {
+  let alertDismiss = (cb) => {
+    cb();
+    alert('Dialog dismissed.');
+  }
+
+  return (
+    <DialogTrigger isDismissable>
+      <ActionButton>Info</ActionButton>
+      {(close) => (
+        <Dialog onDismiss={() => alertDismiss(close)}>
+          <Heading>Version Info</Heading>
+          <Divider />
+          <Content>
+            <Text>
+              Version 1.0.0, Copyright 2020
+            </Text>
+          </Content>    
+        </Dialog>
+      )}
+    </DialogTrigger>
+  );
+}
+```
+
+Note that `onDismiss` is optional. If you don't need to add a `onDismiss` handler to your dismissable Dialog, you may omit
```

Comment:
Should this note be above the example?
Should the example omit the `close()`?

Reply:
A prior dismissable dialog (back in the Examples section) omitted the close function so I can link back to it for clarity. I'm fine with adding a separate example here but the doc is so long with all the examples haha.

Hmm, maybe it would be nice to have the code examples be collapsed by default like the spectrum css markup section, might look into that